### PR TITLE
Event messages are published to Google PubSub

### PR DESCRIPTION
**Pull request by samuel-pt**
## Changes
**setup.py** - Google PubSub module "google-cloud-pubsub" added  as required module
**events/google_injector.py** - Python script to publish messages to Google PubSub
**google_injector_config.sh** - Helper script to setup the environment for PubSub publish

## Steps to setup environment
1. Download service account json from Google Cloud Console. Follow the steps present [here](https://developers.google.com/identity/protocols/application-default-credentials#howtheywork) 
2. Execute `source google_injector_config.sh  "location_to_service_account_file"`. This will install google-cloud-pubsub python module and set GOOGLE_APPLICATION_CREDENTIALS environment variable. Example : `source google_injector_config.sh  "/home/work/v2 pipeline prototype-c222e08d9029.json"`

@Kaitaan @krishnan-chandra

### CODE REVIEW

File: events/google_injector.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+def process_queue(queue, ps_topic,
+                  metrics_client=None):
+    """ Take messages off a queue and send to Google PubSub topic."""
+    while True:
+        message = queue.get()
+        while True:
+            try:
+                ps_topic.publish(message)
+                if metrics_client:
+                    metrics_client.counter("collected.injector").increment()
```
Comment (krishnan-chandra) [inline events/google_injector.py]: Please replace `injector` in these metric names with `google_injector` instead.

File: events/google_injector.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            except GaxError:
+                # In the event of a Google PubSub error in send attempt,
+                #   retry sending after a delay
+                if metrics_client:
+                    metrics_client.counter("injector.pre_send_error").increment()
```
Comment (krishnan-chandra) [inline events/google_injector.py]: Same thing with the metric name

File: events/google_injector.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        max_message_size=MAXIMUM_MESSAGE_SIZE[queue_name],
+    )
+
+    metrics_client = baseplate.make_metrics_client(config)
+    """
+    Assumption: Config will have topic_name for the respective queue_name
+    """
+    topic_name = config["topic." + queue_name]
+
+    while True:
```
Comment (krishnan-chandra) [inline events/google_injector.py]: Is there any reason to have a `while True` here?

File: events/google_injector.py (hunk 1 above)
Comment (samuel-pt) [inline events/google_injector.py]: @krishnan-chandra Replaced injector with google_injector in metric names

File: events/google_injector.py (hunk 3 above)
Comment (samuel-pt) [inline events/google_injector.py]: NotFound exception may arise, when the required topic is not present in Google PubSub. 
This `while True` is just to keep trying until we create the topic manually in Google PubSub. See https://github.com/reddit/event-collector/pull/20/commits/5e7d5fd9902caaf0faeb4128d4471f24da2a1dd8
 
I've also changed warning message, stating that required topic is not present in Google PubSub.

File: google_injector_config.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#/bin/sh
+
+# Install google-cloud-pubsub module
+echo "Installing google-cloud-pubsub python module..."
+pip install --upgrade google-cloud-pubsub
```
Comment (samuel-pt) [inline google_injector_config.sh]: @krishnan-chandra Is it good enough to add the module `google-cloud-pubsub` in `setup.py`

Comment (krishnan-chandra) [inline google_injector_config.sh]: No - we'll need to package up cloud pub sub into a Debian package. I can take care of that.

**Pull request merged**